pkgs/development/libraries/rdkafka/default.nix

```diff
@@ -1,5 +1,17 @@
-{ lib, stdenv, fetchFromGitHub, zlib, zstd, pkg-config, python3, openssl, which, curl }:
-
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  zlib,
+  zstd,
+  pkg-config,
+  python3,
+  openssl,
+  which,
+  curl,
+  cmake,
+  ninja,
+}:
 stdenv.mkDerivation rec {
```

Comment:
I think we should use the `finalAttrs` pattern here.

The `finalAttrs` pattern will let you remove the `rec` keyword ([see its implementation in Nix](https://github.com/NixOS/nix/blob/2d1d81114d72ace89ce08cd3bc93f4eb27a2975d/src/libexpr/nixexpr.cc#L121)).

Why is this pattern preferred to `rec` ?

Let's take this simple code example:

```nix
mkDerivation rec {
   foo = 1;
   bar = foo + 1;
}
```

and then `.overrideAttrs(old: { foo = 2; })`, you'll get `{ foo = 2; bar = 2; }` while with `finalAttrs` pattern, it would work correctly because it's a real fixed point.

Let me share a couple of useful links regarding the `finalAttrs` pattern:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Feel free to reach out if you need some assistance.

Reply:
@drupol I'd like to reach out indeed.

- The "Recent example of implementation" is empty
- Is the finalAttrs pattern even meaningful in this case? If I implement it here and somebody overrides `version`, they'll either see a hash mismatch, and have to override `src` (and the `version` contained therein), or they'll have the source in cache and will silently get the old version source.
- I generally disapprove of this sort of change request given that
  - I'm not the maintainer
  - This is not an `init` PR / the problem was not introduced in this PR
  - The content of the PR is entirely unrelated to the suggested improvements
  - In effect, this feels like "I'm holding your PR hostage until you also do this other work for me"
- Given that this message was written 3 minutes after I opened the PR, I assume it is automated? It would be helpful to at least mark it as such.

Reply:
I'm not holding anything in hostage here, feel free to ignore my suggestion(s) if you want to. I just mentioned this point given the amount of changes you've been making in here. Using the `finalAttrs` pattern is mostly to remove the use of the `rec` keyword, which inherit a very ugly implementation in Nix. Thanks for notifying me about the "Recent example of implementation", I'll update my Github saved replies.

Reply:
> I just mentioned this point given the amount of changes you've been making in here.

I merely applied nixfmt. I suppose I shouldn't have done that, it's unrelated to the content of the PR after all. (Thought I couldn't get a green CI without that, but I guess I can.)

> Github doesn't let me make a fine grained selection of what are the mandatory changes vs the optional changes (yet?)

Would adding a hint to that in the text work? (The GitHub UI is somewhat confusing with these things anyway…)

> which inherit a very ugly implementation in Nix. 

Ah, that's a point I hadn't heard yet, and now I also understand why you linked the nix source.